### PR TITLE
Per 910 - Validate "aws-tde-pg-upgrade" script in EDB-Benchmark platform

### PR DESCRIPTION
**Pull request by dougortiz**
06/01/2023:

- Commented out issues with ansible ‘async’ and ‘poll’
- Fixed key-unwrap-command parameters
- Fixed authentication and execution user account for ‘pg_upgrade’ to execute properly
- Added additional parameters to buildbot interface
  - BENCHMARK_REPO_URL
  - BENCHMARK_REPO_BRANCH
- Fixed issue with TERRAFORM_PLUGIN_CACHE variable
- Added ‘00_validation’ step
- Renamed and renumbered benchmark pipeline steps
- Added ‘[README.md](http://readme.md/)’ file
- Fixed execution ‘pg_upgrade’ execution directory
- Refactored source and target pg_version variables in workflow
- Refactored ‘pg_data’ and ‘pg_wal’ variables for source and target pg_versions
- **Have most benchmark workflow steps working with the exception of the ‘push results’ step because  it is requiring a specific parsing for the output of the upgrade. Paused at this step because was indicated that this step will need team input and refactoring.**

### CODE REVIEW

File: run_benchmark.sh [hunk 1]
```diff
@@ -174,10 +174,10 @@ set_file_permissions() {
...
         # Attempt to handle execute permissions
         # as each directory/step should have a run.sh file
-        if [[ ! -x "$run_script" ]] && [[ ! $(chmod u+x "$run_script") ]]; then
-            echo "Unable to set execution for $run_script" >&2
-            exit 1
-        fi
+        #if [[ ! -x "$run_script" ]] && [[ ! $(chmod u+x "$run_script") ]]; then
+        #    echo "Unable to set execution for $run_script" >&2
+        #    exit 1
+        #fi
```
Comment (bryan-bar) [inline run_benchmark.sh:180]: This following PR changes this to an error message: https://github.com/EnterpriseDB/edb-benchmarks/pull/74

If you could uncomment as it is causing a merge conflict.
```suggestion
        if [[ ! -x "$run_script" ]] && [[ ! $(chmod u+x "$run_script") ]]; then
            echo "Unable to set execution for $run_script" >&2
            exit 1
        fi
```

File: aws-dbt2-rds/validate/run.sh
Comment (bryan-bar) [inline aws-dbt2-rds/validate/run.sh:1]: duplicate file needs to be removed

File: aws-dbt2-rds/validate/run.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   echo "EDB_BENCHMARK_REPO_URL: $EDB_BENCHMARK_REPO_URL";
+   echo "EDB Benchmark Repo URL, cannot be empty!"
+   exit 1
+fi
+
+if [ -z "${EDB_BENCHMARK_REPO_BRANCH}" ]; then
+   echo "EDB_BENCHMARK_REPO_BRANCH: $EDB_BENCHMARK_REPO_BRANCH";
+   echo "EDB Benchmark Repo URL, cannot be empty!"
+   exit 1
+fi
```
Comment (bryan-bar) [inline aws-dbt2-rds/validate/run.sh:34]: Please remove the repo as it is not needed directly by the benchmarks themselves.

File: aws-tde-pg-upgrade/00_validate/run.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -eux
+
+echo "EDB_BENCHMARK_REPO_URL: $EDB_BENCHMARK_REPO_URL"
+echo "EDB_BENCHMARK_REPO_BRANCH: $EDB_BENCHMARK_REPO_BRANCH"
```
Comment (bryan-bar) [inline aws-tde-pg-upgrade/00_validate/run.sh:4]: Please remove this as it is not needed directly by the benchmarks.

File: aws-tde-pg-upgrade/00_validate/run.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+   echo "EDB_BENCHMARK_REPO_URL: $EDB_BENCHMARK_REPO_URL";
+   echo "EDB Benchmark Repo URL, cannot be empty!"
+   exit 1
+fi
+
+if [ -z "${EDB_BENCHMARK_REPO_BRANCH}" ]; then
+   echo "EDB_BENCHMARK_REPO_BRANCH: $EDB_BENCHMARK_REPO_BRANCH";
+   echo "EDB Benchmark Repo URL, cannot be empty!"
+   exit 1
+fi
```
Comment (bryan-bar) [inline aws-tde-pg-upgrade/00_validate/run.sh:28]: Please remove this as it is not needed directly by the benchmarks.

We also need a validation for the repo credential variables:
```shell
EDB_REPO_USERNAME
EDB_REPO_PASSWORD
```

File: aws-tde-pg-upgrade/05_push_results/run.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -eux
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# Push pgbench files to the S3 bucket
+
+# Copy collected data and generated data & charts
+cp -r ${SCRIPT_DIR}/../03_execute/benchmark_data ${SCRIPT_DIR}/report-data
```
Comment (bryan-bar) [inline aws-tde-pg-upgrade/05_push_results/run.sh:8]: The previous directory index needs to be updated until we have a variable that holds all the results.

```suggestion
cp -r ${SCRIPT_DIR}/../04_execute/benchmark_data ${SCRIPT_DIR}/report-data
```

File: aws-tde-pg-upgrade/environment [hunk 6]
```diff
@@ -2,7 +2,7 @@ TERRAFORM_PROJECT_PATH=
 SSH_USER="rocky"
 EDB_REPO_USERNAME=
 EDB_REPO_PASSWORD=
```
Comment (bryan-bar) [inline aws-tde-pg-upgrade/environment:4]: This should be added to the validation.

File: aws-tde-pg-upgrade/00_validate/run.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+   echo "tpcc_warehouse: $tpcc_warehouse";
+   echo " Warehouses, cannot be lower than 25!"
+   exit 1
+fi
+
+if (( $(echo "$tpcc_loader_vusers< 25" | bc -l) )); then
+   echo "tpcc_loader_vusers: $tpcc_loader_vusers";
+   echo "Hammer DB number of virtual users, cannot be lower than 25!"
+   exit 1
+fi
```
Comment (bryan-bar) [inline aws-tde-pg-upgrade/00_validate/run.sh:16]: Please update these to use UPPER case.

**Review by bryan-bar — CHANGES_REQUESTED** [review]
Please review the comments below:

File: run_benchmark.sh (hunk 1 above)
Comment (dougortiz) [reply run_benchmark.sh:180]: Un-commented

File: aws-tde-pg-upgrade/00_validate/run.sh (hunk 3 above)
Comment (dougortiz) [reply aws-tde-pg-upgrade/00_validate/run.sh:4]: Removed

File: aws-tde-pg-upgrade/05_push_results/run.sh (hunk 5 above)
Comment (dougortiz) [reply aws-tde-pg-upgrade/05_push_results/run.sh:8]: Updated

File: aws-tde-pg-upgrade/00_validate/run.sh (hunk 4 above)
Comment (dougortiz) [reply aws-tde-pg-upgrade/00_validate/run.sh:28]: Added to validation

File: aws-tde-pg-upgrade/00_validate/run.sh (hunk 7 above)
Comment (dougortiz) [reply aws-tde-pg-upgrade/00_validate/run.sh:16]: Updated to UPPER case

**Pull request closed without merging**